sky/execution.py

```diff
@@ -377,31 +378,52 @@ def spot_launch(
               f'{colorama.Style.RESET_ALL}')
         return
 
-    # TODO(zhwu): Refactor the Task (as Resources), so that we can enforce the
-    # following validations.
-    # Check the file mounts in the task.
-    # Disallow all local file mounts (copy mounts).
+    # Translate the workdir and local file mounts to cloud file mounts.
+    new_storage_mounts = {}
+    run_id = common_utils.get_run_id()[:8]
+
     if task.workdir is not None:
-        with ux_utils.print_exception_no_traceback():
-            raise exceptions.NotSupportedError(
-                'Workdir is currently not allowed for managed spot jobs.\n'
-                'Hint: use Storage to auto-upload the workdir to a cloud '
-                'bucket.')
+        workdir = task.workdir
+        task.workdir = None
+        bucket_name = spot.constants.SPOT_WORKDIR_BUCKET_NAME.format(
+            username=getpass.getuser(),
+            hash=run_id,
+        )
+        new_storage_mounts[backend_utils.SKY_REMOTE_WORKDIR] = (
+            storage.Storage.from_yaml_config({
+                'name': bucket_name,
+                'source': workdir,
+                'persistent': False,
+                'mode': 'COPY',
+            }))
+        logger.info(
+            f'Translated workdir {workdir} to cloud storage {bucket_name}.')
```

Comment:
How about 
> Workdir {workdir} will be synced to cloud storage {bucket_name}. See `sky storage ls`.

Not too sure about how to hint the last part as at this point it won't show this Storage object?

Reply:
Good point! I moved the `sky storage ls` hint before the storage upload, so that the user knows that the storage will be shown in the table after the source is uploaded.